src/HypothesisTests.jl

```diff
@@ -89,14 +86,17 @@ function Base.show{T<:HypothesisTest}(io::IO, test::T)
     println(io)
 
     # test summary
-    p, tail = pvalue(test)
+    p = pvalue(test)
     outcome = if p > 0.05 "fail to reject" else "reject" end
+    tail = used_tail_of_distribution(test)
     println(io, "Test summary:")
     println(io, "    outcome with 95% confidence: $outcome h_0")
     if tail == :both
         println(io, "    two-sided p-value:           $p")
-    else
+    elseif tail == :left || tail == :right
         println(io, "    one-sided p-value:           $p")
+    else
+        println(io, "    ??-sided p-value:            $p")
```

Comment:
I have changed it to just say `p-value` if a test does not set `tail_of_distribution()`. Btw, feel free to propose a different name for that function.

All tests pass locally. Let met know if I should do anything else. Thanks!